### ISSUE TITLE
[azeventhubs] Address Sandeep's feedback

### ISSUE THREAD

**Issue by richardpark-msft** [Azure/azure-sdk-for-go#19436]
- [x] Logs mention 'credits', but that term can be confusing to users who might think it's related to billing or something similar.
- [x] Create a whole file example and produces and consumes so we can indicate that partition IDs exist and need to match up
- [x] Create a whole file example for peeking at an event hub.
- [x] Batch sizing + waiting can be potentially confusing since we don't early-exit on first receive. So if you ask for 100 events we'll block even if we got a few since the batch isn't full and you haven't canceled your context. How do we properly explain this.
- [x] Document more about batching (@sandeep-sen will add more clarification)
- [ ] Maybe an HTTP app would be good while file sample.
- [x] "Response router has shut down"

CC: @sandeep-sen

**Comment by richardpark-msft** [comment]
@sandeep-sen, I did some updates to the docs in https://github.com/Azure/azure-sdk-for-go/pull/19533, take a look and let me know if it helps at all with the things we discussed.

Instead of creating a single file example for sending/receiving I kept it separate, but I changed the receiving side to receive from _all_ partitions, instead of just one. I think this'll help drive home the concept that partitions exist, and when the run the program, it'll also be more obvious that their messages are getting routed.

I also added in some more docs to the producing sample, to talk more about what the effects are of the different settings with relation to partition targeting.

I did a teeny bit with the Processor sample. I think that one is a bit complex, and could be factored more but I wanted to take a sceond more on it. I did update the docs in general to try to make it simpler to understand the flow, but I'll have to take another run at it, probably during the Nov-Dec timeframe.
